android/cli/commands/_build.js

```diff
@@ -2825,6 +2825,19 @@ AndroidBuilder.prototype.copyResources = function copyResources(next) {
 			this.encryptJS && jsFilesToEncrypt.push('_app_props_.json');
 			delete this.lastBuildFiles[appPropsFile];
 
+			// write the env variables file
+			const envVarsFile = path.join(buildAssetsPath, '_env_.json');
+			Object.keys(this.tiapp.properties).forEach(function (prop) {
+				props[prop] = this.tiapp.properties[prop].value;
+			}, this);
+			fs.writeFileSync(
+				envVarsFile,
+				// for non-development builds, DO NOT WRITE OUT ENV VARIABLES TO APP
```

Comment:
Maybe i'm missing something here, but why is this disabled for production builds? What if i want to use environment variables to control certain things in my app?

Also, the `deployType` for device builds is `test` which would make this unreliable when switching between sim and device builds during development.

Reply:
So my basic concern here was that people tend to forget and have some identifying information, api keys, passwords, whatever in their environment variables and sneaking those by default into the production/released build of their app seems like a security accident waiting to happen. I think for production builds developers will likely want to be much more particular about what actually goes in. For those cases, I suspect they'd want an approach more like https://github.com/motdotla/dotenv

But, you're saying when we target a physical device from `ti build` the `deployType === 'test'`? If os, then I guess I can just change the guard here to be `deployType !== 'production'`